pkg/apply/format.go

```diff
@@ -93,6 +93,68 @@ func FormatSettingsDiff(
 	return string(bytes.TrimRight(buf.Bytes(), "\n")), nil
 }
 
+// FormatSettingsDiffTracker formats the settings diffs as a ChangesTracker object instead of a table
+func FormatSettingsDiffTracker(
+	topicName string,
+	topicSettings config.TopicSettings,
+	configMap map[string]string,
+	diffKeys []string,
+) (*ChangesTracker, error) {
+
+	configEntries := make([]ConfigEntryChanges, 0)
+	for _, diffKey := range diffKeys {
+		configValueStr := configMap[diffKey]
+
+		var valueStr string
+		var err error
+
+		if topicSettings.HasKey(diffKey) {
+			valueStr, err = topicSettings.GetValueStr(diffKey)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		configEntries = append(configEntries, ConfigEntryChanges{
+			Name:    diffKey,
+			Current: configValueStr,
+			Updated: valueStr,
+		})
+	}
+
+	return &ChangesTracker{
+		Topic:              topicName,
+		NumPartitions:      IntValueChanges{},
+		ReplicationFactor:  IntValueChanges{},
+		ReplicaAssignments: []ReplicaAssignmentChanges{},
+		ConfigEntries:      configEntries,
+		MissingKeys:        []string{},
+		Action:             ActionEnumUpdate,
```

Comment:
I'd assume you are going to fill in the other fields that are initialized empty here.

Reply:
commented out the fields which will be implemented in future PRs